fd/diff.go

```diff
@@ -175,15 +174,14 @@ func Gradient(dst []float64, f func([]float64) float64, x []float64, settings *S
 	for i := 0; i < nWorkers; i++ {
 		go func(sendChan <-chan fdrun, ansChan chan<- fdrun, quit <-chan struct{}) {
 			xcopy := make([]float64, len(x))
-			copy(xcopy, x)
 			for {
 				select {
 				case <-quit:
 					return
 				case run := <-sendChan:
+					copy(xcopy, x)
 					xcopy[run.idx] += run.pt.Loc * step
```

Comment:
Note that we should be smarter about this - if the formula contains the origin, we are re-evaluating the function there len(x) times.